gprotorch/dataloader/reaction_loader.py

```diff
@@ -0,0 +1,130 @@
+        :type nBits: int
+        """
+
+        valid_representations = [
+            "ohe",
+            "rxnfp",
+            "drfp",
+            "bag_of_smiles",
+        ]
+
+        if representation == "ohe":
+            self.features = one_hot(self.features)
+
+        elif representation == "rxnfp":
+            self.features = rxnfp(self.features.to_list())
+
+        elif representation == "drfp":
+            self.features = drfp(self.features.to_list(), nBits=nBits)
+
+        elif representation == "bag_of_smiles":
+            self.features = bag_of_characters(self.features.to_list())
+
+        else:
+            raise Exception(
+                f"The specified representation choice {representation} is not a valid option. "
+                f"Choose between {valid_representations}."
+            )
+
+    def load_benchmark(self, benchmark, path):
+
+        """Loads features and labels from one of the included benchmark datasets
+                and feeds them into the DataLoader.
+
+        :param benchmark: the benchmark dataset to be loaded, one of
+            ``[DreherDoyle, SuzukiMiyaura, DreherDoyleRXN, SuzukiMiyauraRXN]``
+              RXN suffix denotes that csv file contains reaction smiles in a dedicated column.
+        :type benchmark: str
+        :param path: the path to the dataset in csv format
+        :type path: str
+        """
+
+        benchmarks = {
+            "DreherDoyle": {
+                "features": ["ligand", "additive", "base", "aryl halide"],
+                "labels": "yield",
+            },
+            "DreherDoyleRXN": {"features": "rxn", "labels": "yield"},
+            "SuzukiMiyaura": {
+                "features": [
+                    "reactant_1_smiles",
+                    "reactant_2_smiles",
+                    "catalyst_smiles",
+                    "ligand_smiles",
+                    "reagent_1_smiles",
+                    "solvent_1_smiles",
+                ],
+                "labels": "yield",
+            },
+            "SuzukiMiyauraRXN": {"features": "rxn", "labels": "yield"},
+        }
+
+        if benchmark not in benchmarks.keys():
+
+            raise Exception(
+                f"The specified benchmark choice ({benchmark}) is not a valid option. "
+                f"Choose one of {list(benchmarks.keys())}."
+            )
+
+        else:
+
+            df = pd.read_csv(path)
+            # drop nans from the datasets
+            nans = df[benchmarks[benchmark]["labels"]].isnull().to_list()
+            nan_indices = [nan for nan, x in enumerate(nans) if x]
+            self.features = df[benchmarks[benchmark]["features"]].drop(
+                nan_indices
+            )
+            self.labels = (
+                df[benchmarks[benchmark]["labels"]]
+                .dropna()
+                .to_numpy()
+                .reshape(-1, 1)
+            )
+
+
+if __name__ == "__main__":
```

Comment:
This is a test for the loader functionality? Might be good to have this as an example e.g. line 205 here:

https://github.com/pytorch/botorch/blob/main/botorch/models/gp_regression.py